frontend/src/redux/sagas/restaurants/index.js

```diff
@@ -2,26 +2,43 @@ import { put, takeEvery } from 'redux-saga/effects'
 import axios from 'axios';
 
 import { types, restaurantsActions } from '../../reducers/restaurants';
-import { BACKEND_SERVICE_PATH } from '../../constants';
+import { GOOGLE_API_PATH } from '../../constants';
+import { API_KEY } from '../../../properties'
 
-const API_PATH = `${BACKEND_SERVICE_PATH}/restaurants`;
+const HEADER_CONTENT_TYPE = 'Content-Type';
+const HEADER_API_KEY = 'X-Goog-Api-Key';
+const HEADER_FIELD_MASK = 'X-Goog-FieldMask';
 
-export function* callGetRestaurants() {
+export function* callGetRestaurants({ textQuery }) {
     try {
-        const { data } = yield axios.get(API_PATH);
+        const { data } = yield axios.post(GOOGLE_API_PATH + "places:searchText",
+            { 'textQuery': textQuery },
+            { headers: {
+                [HEADER_CONTENT_TYPE]: 'application/json',
+                [HEADER_API_KEY]: API_KEY,
+                [HEADER_FIELD_MASK]: 'places.displayName,places.formattedAddress,places.id'
+            }}
+        );
         yield put(restaurantsActions.successfulGetRestaurantsRequest(data));
     } catch (err) {
-        yield put(restaurantsActions.failedGetRestaurantsRequest('Failed getting restaurants'));
+        yield put(restaurantsActions.failedGetRestaurantsRequest('Failed getting restaurants from Google'));
     }
 }
 
 export function* callGetRestaurantById({ restaurantId }) {
```

Comment:
Something to consider: The information returned by [text search](https://developers.google.com/maps/documentation/places/web-service/text-search) is the same as the one returned by [place details API](https://developers.google.com/maps/documentation/places/web-service/place-details)  called in this function. We can potentially minimize cost here (at potentially increase in latency) by passing API data onwards without needing a second API call in every case for each restaurant selected off of the restaurants page.

Also to consider from @NickPriv: If we click on specific restaurant from restaurants page, we should already have a way to get restaurant-specific details from place API, but if we share the link to someone else or go direct to the link without going through the search bar, we will not have any API data and will need to make a new API request to get data.

Reply:
Action Item: Create an improvement issue for this.

Reply:
Created the issue: https://github.com/NickPriv/FryRank/issues/31